### PR TITLE
Make vector classes templates

### PR DESCRIPTION
**Pull request by AndreasLrx**
Vectors can be of any arithmetic type.

Sorry for this late refactoring, I need integral values to work with the game map.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -68,7 +68,7 @@ static void drawFrame(void *arg)
     raylib::core::Window::clear();
     {
         raylib::core::scoped::Mode3D mode3D(*camera);
-        // testingModel.draw(pos, rotationAxis, -90, scale, raylib::core::Color::RED);
+        testingModel.draw(pos, rotationAxis, -90, scale, raylib::core::Color::RED);
```
Comment (aurelenc) [inline src/main.cpp:71]: Forgot to remove debug ?

**Review by aurelenc — APPROVED** [review]

File: src/main.cpp (hunk 1 above)
Comment (AndreasLrx) [reply src/main.cpp:71]: It will be removed later anyway when merging the branch on the game or on the Worlds.

Comment (AndreasLrx) [reply src/main.cpp:71]: (With this line commented it made the CI failed because of the unused values above)

**Review by MisterPeModder — APPROVED** [review]
LGTM

**Pull request merged**